src/client/client.cc

```diff
@@ -186,7 +186,7 @@ namespace Pistache::Http::Experimental
         void registerPoller(Polling::Epoll& poller) override;
         void unregisterPoller(Polling::Epoll& poller) override;
 
-        Async::Promise<void> asyncConnect(std::shared_ptr<Connection> connection,
+        Async::Promise<void> asyncConnect(const std::shared_ptr<Connection>& connection,
```

Comment:
Is this related to the modified lambda captures? If not, please either split this in a separate commit explaining why this is needed or drop this change

Reply:
Hm, no, it's just an error. When I zoned out for a little and just started pressing "yes" to the CLion suggestions to fix things.

Reply:
It is not an error. It is only chasing the reference counter of `std::shared_ptr` more than needed. Best would be to use rvalue-ref or to move it into the `ConnectionEntry`.